### ISSUE TITLE
Devstack mounts not quite right

### ISSUE THREAD

**Issue by markdryan** [intel/ccloudvm#77]
I don't think the way we're mounting paths inside the devstack workload is correct.

Currently, the workload provides the option of mounting a host devstack directory inside the guest.  There are a couple of problems with this.

1. The mounting doesn't work very well due to #50.  stack.sh fails for me when I try to create a devstack-xenial workload that mounts my host's devstack directory.  I think it fails as the guest cannot write to the devstack directory.
2. The devstack directory should be local to the guest as it contains guest local settings.

Also, what we should really be sharing is /opt/stack, as far as I can tell.  Doing so would allow users to edit OS code on the host but deploy it in a guest.  At least I hope it would.  Please correct me if I'm wrong.

How about we change things a little bit.

1. Get rid of the devstack mount.
2. Always clone devstack and create the local.conf as we currently do.
3. Then we exit with a message, i.e., we don't actually run stack.sh inside the cloud-init file at all.  Instead we print out a message explaining what do to, e.g.,

```
devstack initialised with a default local.conf

[[local|localrc]]
HOST_IP=127.3.232.1
ADMIN_PASSWORD=secret
DATABASE_PASSWORD=$ADMIN_PASSWORD
RABBIT_PASSWORD=$ADMIN_PASSWORD
SERVICE_PASSWORD=$ADMIN_PASSWORD

To start devstack with these defaults type

$ ccloudvm run INSTANCE-NAME "cd devstack && ./stack.sh"

To start with a custom local.conf

$ ccloudvm copy local.conf devstack/local.conf
$ ccloudvm run INSTANCE-NAME "cd devstack && ./stack.sh"

To access horizon point your host's browser at http://127.3.232.1:55580.
If you're using proxies ensure that 127.3.232.1 is in your no_proxy settings.
```

**Comment by markdryan** [comment]
Of course, there's one problem with the above proposal.  $ ccloudvm copy local.conf devstack/local.conf isn't actually implemented.  This is bug #5.  Should be easy to add though.

**Comment by markdryan** [comment]
@ganeshmaharaj Any thoughts on all this?

**Comment by markdryan** [comment]
Also we could only print out
```
To access horizon point your host's browser at http://127.3.232.1:55580.
If you're using proxies ensure that 127.3.232.1 is in your no_proxy settings.
```
If the user is using local IP addresses for the guest VM.

**Comment by ganeshmaharaj** [comment]
@markdryan the repos inside /opt/stack are populated by devstack's stack.sh and OS users until now expect that to be completely throwaway (in the sense, they never expect to have a copy of repos in their host's /opt/stack). I am sure they can keep a clone on their host if they choose to, but i am not a 100% sure of all the assumptions devstack scripts operate upon which might/might not mess up the contents of /opt/stack. I can check with other users and see if they would have some preference on this. 

About the issue of not able to write into the devstack repo, i thought it worked for me, but i can check on my side to see if i hit any other issues. We could also mount the basedir of the devstack repo and that should let us write to devstack folder if that is an issue.

**Comment by ganeshmaharaj** [comment]
@markdryan after our conversation, maybe best to avoid mounting devstack itself, instead copy the local.conf to devstack and then initiate stack.sh post copy.

**Comment by markdryan** [comment]
Fixed by PR #80
